arch/arc/core/fatal.c

```diff
@@ -34,3 +34,12 @@ FUNC_NORETURN void z_arch_syscall_oops(void *ssf_ptr)
 	z_arc_fatal_error(K_ERR_KERNEL_OOPS, ssf_ptr);
 	CODE_UNREACHABLE;
 }
+
+FUNC_NORETURN void z_arch_system_halt(unsigned int reason)
+{
+	ARG_UNUSED(reason);
+
+	__asm__("flag 1");
```

Comment:
Not sure flag is the right solution here:
- calling from normal kernel mode will trigger a Privilege Violation exception
- calling from any user mode, flag 1 will be ignored

What privilege levels/modes can call this function?
A BRK instruction might be better, and/or and endless loop after the instruction?

Reply:
1. What do you mean saying "normal kernel mode"? Is that a kernel mode as opposed to being in secure mode? Then we may wrap it in corresponding defines and at least w/o secure mode we may execute `flag 1`.

2. Correct, but the question is if it's even possible to reach `z_arch_system_halt()` from user-mode?

In general my rationale for implementation of this function was an ability to get simulator or debugger execution stop. I.e. instead of keeping target looping infinitely and debugger/simulator running that long as well I'd prefer them to exit and so we'll be able to capture that event with external tooling.

Moreover I had a naive hope that whenever we go to the end of `main()` we'll eventually end-up in exactly that `z_arch_system_halt()`. But I was not right... but that's a completely separate discussion I guess.

Reply:
Yes, I meant 'normal' as opposed to 'secure' mode. We've initial support for this SecureShield feature, want to prevent issues with this in the future. I think that instead of flag, you could use a brk instruction to have the same result of exiting the simulator? That works in both normal and secure mode. And, if enabled by the debugger, also in user mode (debugger has to set some bit in the debug control register).

Reply:
OK, makes sense. Will send a re-spin now.